### ISSUE TITLE
Unable to disable self-update feature

### ISSUE THREAD

**Issue by lcheylus** [nextest-rs/nextest#1204]
I'm building cargo-nextest current version on OpenBSD-current/amd64, using my patch for `libc` crate (add missing `waitid` function, see https://github.com/rust-lang/libc/pull/3490) => **build OK**

- I'm trying to disable `self-update` feature according to https://nexte.st/book/updating.html for packaging the tool.
```shell
$ cargo build -v --no-default-features --features default-no-update
```
But the **"self-update" feature is not disabled** with this build:
```shell
$ target/debug/cargo-nextest nextest self update --help
Download and install updates to nextest

This command checks the internet for updates to nextest, then downloads and installs them if an update is available.

Usage: cargo nextest self update [OPTIONS]

$ target/debug/cargo-nextest nextest self update
info: checking for self-updates
info: cargo-nextest is already at the latest version: 0.9.66
```

- If I try to remove "self-update" feature from `default` features in `cargo-nextest/Cargo.toml`:
```toml
[features]
default = []
```
the **"self-update" is correctly disabled** after build:
```shell
$ target/debug/cargo-nextest nextest self update --help
This version of nextest does not have self-update enabled

Always exits with code 93 (SELF_UPDATE_UNAVAILABLE).

Usage: cargo nextest self update [OPTIONS]

$ target/debug/cargo-nextest nextest self update
info: this version of cargo-nextest cannot perform self-updates
(hint: this usually means nextest was installed by a package manager)
```

I don't understand why "self-update" feature is not disabled using `--no-default-features --features default-no-update`.

**Comment by sunshowers** [comment]
Thanks, fixed in 770b6cc.

**Comment by lcheylus** [comment]
OK, build the last version with `--no-default-features --features default-no-update`

```
$ ./target/debug/cargo-nextest nextest self update --help
This version of nextest does not have self-update enabled

Always exits with code 93 (SELF_UPDATE_UNAVAILABLE).
(...)

./target/debug/cargo-nextest nextest self update
info: this version of cargo-nextest cannot perform self-updates
(hint: this usually means nextest was installed by a package manager)
```

Thanks
